Sources/SwiftASN1/Basic ASN1 Types/GeneralizedTime.swift

```diff
@@ -228,6 +287,16 @@ extension GeneralizedTime: Comparable {
         if lhs.hours < rhs.hours { return true } else if lhs.hours > rhs.hours { return false }
         if lhs.minutes < rhs.minutes { return true } else if lhs.minutes > rhs.minutes { return false }
         if lhs.seconds < rhs.seconds { return true } else if lhs.seconds > rhs.seconds { return false }
-        return lhs.fractionalSeconds < rhs.fractionalSeconds
+        if lhs.fractionalSeconds < rhs.fractionalSeconds { return true } else if lhs.fractionalSeconds > rhs.fractionalSeconds { return false }
+
+        for (lhsByte, rhsByte) in zip(lhs.rawFractionalSeconds, rhs.rawFractionalSeconds) {
+            if lhsByte != rhsByte {
+                return lhsByte < rhsByte
+            }
+        }
+
+        // Since the above `zip` iteration stops at the length of the shorter `Sequence`, finally,
+        // compare the length of the two `Sequence`s.
+        return lhs.rawFractionalSeconds.count < rhs.rawFractionalSeconds.count;
```

Comment:
Related to the discussion above about the source of truth. I think `rawFractionalSeconds` should be the source of truth and `fractionalSeconds` is just a cached/derived value from `rawFractionalSeconds` as it will always be more precise. 

Therefore we might want to get rid of the `lhs.fractionalSeconds < rhs.fractionalSeconds` check. I'm not certain though as it might be a performance improvement. Is it guaranteed that there isn't a conflict between these two values? I guess so but I'm not sure. Either way, worth a comment that this is just an additional check to improve performance.

With that being said, how expensive is a double comparison compared to iterating over an array if the array is smallish?

Reply:
A double comparison is _very_ cheap compared to the array iteration. As close to free as makes no odds for us here.